packages/react/src/checkbox/Checkbox.css.ts

```diff
@@ -0,0 +1,111 @@
+// import { theme } from "../styles";
+import { theme } from "../styles";
+import { recipe, style } from "../vanilla-extract";
+
+export const iconC = recipe({
+  base: [
+    style({
+      display: "none",
+      selectors: {
+        '[data-state="checked"] &': {
+          display: "block",
+        },
+      },
+    }),
+  ],
+});
+
+export const iconUC = recipe({
+  base: [
+    style({
+      display: "none",
+      selectors: {
+        '[data-state="unchecked"] &': {
+          display: "block",
+        },
+      },
+    }),
+  ],
+});
+
+export const iconI = recipe({
+  base: [
+    style({
+      display: "none",
+      selectors: {
+        '[data-state="indeterminate"] &': {
+          display: "block",
+        },
+      },
+    }),
+  ],
+});
+
+export const indicator = recipe({
+  base: [
+    style({
+      alignItems: "center",
+      background: theme.colors["bg.brand.solid"],
+      borderRadius: theme.borderRadius["xs"],
+      color: "black",
+      display: "flex",
```

Comment:
We can use sprinkles if available where we aren't overriding them in selectors. 
Keeping `backgroundColor` in style as we are overriding in selectors.